### PR TITLE
feat/RSS-ECOMM-2_11 Integrate the registration form v2

### PR DESCRIPTION
**Pull request by AndreiMakhnach2209**
### What type of PR is this?

- [ ] 🌟 Feature
- [x] 🛠 Bug Fix
- [ ] ⚙️ Refactor
- [ ] 📖 Documentation

### Description

- **Brief Overview:**
  
    Fix creating new Customer
- **Implementation Approach:**
  

### Additional Information

- **Screenshots/Links:**
  
- [ ] Related Issues:

### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [ ] 📝 I have commented my code, particularly in hard-to-understand areas.
- [ ] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.

### CODE REVIEW

File: src/api/API.tsx [hunk 1]
```diff
@@ -22,29 +23,41 @@ class API {
...
+              params: {
+                grant_type: "password",
+                scope: process.env.CTP_SCOPES,
+                username: customerData.email,
+                password: customerData.password,
+              },
+              auth: {
+                username: process.env.CTP_CLIENT_ID ?? "",
+                password: process.env.CTP_CLIENT_SECRET ?? "",
+              },
```
Comment (Oleg-Melnikow) [inline src/api/API.tsx]: Can we avoid duplication of code in this case and can we create variable for it, for example `auth`

File: src/api/API.tsx [hunk 2]
```diff
@@ -22,29 +23,41 @@ class API {
...
-          auth: {
-            username: process.env.CTP_CLIENT_ID ?? "",
-            password: process.env.CTP_CLIENT_SECRET ?? "",
-          },
-        }
-      );
+      const response = customerData
+        ? await axios.post(
+            `${process.env.CTP_AUTH_URL}/oauth/${process.env.CTP_PROJECT_KEY}/customers/token`,
+            null,
```
Comment (Oleg-Melnikow) [inline src/api/API.tsx:35]: Can we use in this place `this.instance` instead of `axios`?

**Review by Oleg-Melnikow — COMMENTED** [review]
Good job! Check comment please.

**Review by JeckJonnyQ — APPROVED** [review]

File: src/api/API.tsx (hunk 2 above)
Comment (AndreiMakhnach2209) [reply src/api/API.tsx:35]: `This.instance` uses the base URL for API requests, and here the authorization URL is needed.

**Review by Oleg-Melnikow — APPROVED** [review]
Nice job!

**Pull request merged**